src/main/java/gift/main/dto/OptionListRequest.java

```diff
@@ -0,0 +1,29 @@
+package gift.main.dto;
+
+import gift.main.Exception.CustomException;
+import gift.main.Exception.ErrorCode;
+import jakarta.validation.Valid;
+import jakarta.validation.constraints.NotEmpty;
+import org.springframework.validation.annotation.Validated;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+@Validated
+public record OptionListRequest(
+        @NotEmpty @Valid List<OptionRequest> optionRequests) {
```

Comment:
```suggestion
    @Size(min = 1)
    List<@Valid OptionRequest> optionRequests) {
```

`@Valid` 를 통해서 내부 검증해볼 수 있을 것 같아요.

```java
        if (name.isBlank()) {
            throw new CustomException(ErrorCode.EMPTY_OPTION);
        }
        if (name.length() > 50) {
            throw new CustomException(ErrorCode.INVALID_OPTION_NAME_LENGT);
        }
```
내부 validation 메서드도 어노테이션으로 검증해볼 수 있을거에요!